File: 프로그래머스/unrated/138477. 명예의 전당 （1）/명예의 전당 （1）.java
import java.util.*;

class Solution {
    public int[] solution(int k, int[] score) {
        int[] answer = new int[score.length];
        int min = score[0];
        ArrayList<Integer> arr = new ArrayList<Integer>();
        if (k >= score.length) {
            for (int i = 0; i < score.length; i++) {
            if (min > score[i]) {
                min = score[i];
            }
            answer[i] = min;
        }
        } else {
        for (int i = 0; i < k; i++) {
            if (min > score[i]) {
                min = score[i];
            }
            answer[i] = min;
            arr.add(score[i]);
        }
        
        for (int i = k; i < score.length; i++) {
            arr.add(score[i]);
            Collections.sort(arr);
            answer[i] = arr.get(i-k+1);
        }
        }
        return answer;
    }
}

File: 프로그래머스/1/138477. 명예의 전당 （1）/명예의 전당 （1）.py
def solution(k, score):
    answer = []
    record = []
    for i in score:
        if len(record) >= k:
            if min(record) < i:
                record.sort()
                record[0] = i
                answer.append(min(record))
            else:
                answer.append(min(record))
        else:
            record.append(i)
            answer.append(min(record))
    return answer